joomla/joomla-cms
# Media fields accessibility

Pull Request for Issue https://github.com/joomla/joomla-cms/issues/27571 .

### Summary of Changes
I have integrated the possibility to enter an alternative text for a media field. This PR is a draft because it affects many other components. I haven't checked them all yet. I would like to know your opinion first.



### Testing Instructions

1. Apply this patch. The commands `npm ci` or/and `composer i` are not necessary.
2. Create two custom fields of type `media` in the content component. Choose `yes` for the `Alternative text for image` option and `no` for the other.
![Articles  Fields   test   Administration(2)](https://user-images.githubusercontent.com/9974686/73128476-41ecb200-3fd0-11ea-8cb1-7a69ace00087.png)
![Articles  New Field   test   Administration](https://user-images.githubusercontent.com/9974686/73128475-41ecb200-3fd0-11ea-9ab8-f6b3ad959423.png)
3. Edit an article that contains the two custom fields. You only see the text field where you chose to enter an alternative text.
![Articles  New   test   Administration](https://user-images.githubusercontent.com/9974686/73128514-c2131780-3fd0-11ea-911c-ee5f83ab5706.png)
4. Look at the output in the front end. Once you see the alternative text you entered. The image, for which you could not enter any text, is also a11y compliant. Here is an empty alt text inserted.
5. Since the media field is used in many places in Joomla, many places have to be adjusted. Therefore, this first 4 steps are not enough. Look for different places where the media field is used. For example, create a banner module with pictures, create a contact with a picture or an article with a picture. Always look at the output in the front end. Make sure the image is saved and dispayed correctly. Write it here in the issue if you find a place that is not correct.
6. Make sure that the text field for entering an alternative text is not displayed in other fields than an custom field. So far, this alternative text has not yet been forwarded to the output in the front end, except for a custom field. Therefore, this would confuse an user. Usually there is an extra text field to use the alternative text. 
![Articles  New   test   Administration(1)](https://user-images.githubusercontent.com/9974686/73128762-78c4c700-3fd4-11ea-80d2-28c10b04d2b5.png)
Perhaps we should consider doing this uniformly in the future via the media field.








### Documentation Changes Required
https://docs.joomla.org/J3.x:Adding_custom_fields/Media_Field
